src/index.js

```diff
@@ -1,22 +1,21 @@
 import React from "react";
 import ReactDOM from "react-dom";
-import { createStore, applyMiddleware } from "redux";
 import { Provider } from "react-redux";
-import createSagaMiddleware from "redux-saga";
-import { composeWithDevTools } from "redux-devtools-extension";
+import { buildRouter } from "@salsita/react-router";
+import { buildStore } from "@salsita/react-core";
 
 import { Root } from "modules/root/components/root";
 import { rootReducer } from "modules/root/root-reducer";
 import { rootSaga } from "modules/root/root-saga";
+import * as routes from "router/routes";
 
-const sagaMiddleware = createSagaMiddleware();
+const router = buildRouter([routes.USER_LIST, routes.USER_DETAIL], {
+  defaultRoute: routes.USER_LIST.name
+});
 
-const store = createStore(
-  rootReducer,
-  composeWithDevTools(applyMiddleware(sagaMiddleware))
-);
+router.start();
```

Comment:
I think it's better to start the router later when the store is ready. Not sure if it correctly propagates the first transition to the store now. Also, you can render the app when the router is ready if you use a callback (see the arrow function). I would rather do it in the following order:

```ts
const router = buildRouter(...);
const store = buildStore(...);

router.start(() => ReactDOM.render(
  <Provider store={store}>
    <Root />
  </Provider>,
  document.getElementById("root")
));